File: uplift-gradle-plugins/src/main/java/com/github/kjetilv/uplift/plugins/UpliftBootstrapTask.java
package com.github.kjetilv.uplift.plugins;

import org.gradle.api.provider.Property;
import org.gradle.api.tasks.CacheableTask;
import org.gradle.api.tasks.OutputFile;

import java.nio.file.Path;

/** Bootstraps the CDK toolkit stack in the target account and region. */
@CacheableTask
public abstract class UpliftBootstrapTask extends UpliftLambdaZipTask {

    @OutputFile
    public abstract Property<Path> getTemplate();

    @Override
    protected void perform() {
        collectLambdaZips();
        cdk().run("cdk bootstrap " + profileOption() + " aws://" + getAccount().get() + "/" + getRegion().get());
    }
}

File: uplift-gradle-plugins/src/main/kotlin/com/github/kjetilv/uplift/plugins/UpliftBootstrapTask.kt
package com.github.kjetilv.uplift.plugins

import org.gradle.api.provider.Property
import org.gradle.api.tasks.CacheableTask
import org.gradle.api.tasks.OutputFile
import org.gradle.process.ExecOperations
import java.nio.file.Path
import javax.inject.Inject

@CacheableTask
abstract class UpliftBootstrapTask @Inject constructor(execOperations: ExecOperations) :
    UpliftLambdaZipTask(execOperations) {

    @get:OutputFile
    abstract val template: Property<Path>

    override fun perform() {
        collectLambdaZips()
        bootstrapCdk()
    }

    private fun bootstrapCdk() {
        runCdk("cdk bootstrap ${profileOption()} aws://${account.get()}/${region.get()}")
    }
}
